rqt_image_view/src/rqt_image_view/ratio_layouted_frame.cpp

```diff
@@ -77,26 +77,33 @@ void RatioLayoutedFrame::resizeToFitAspectRatio()
   QRect rect = contentsRect();
 
   // reduce longer edge to aspect ration
-  double width = double(rect.width());
-  double height = double(rect.height());
-  if (width * aspect_ratio_.height() / height > aspect_ratio_.width())
+  double width = outer_layout_->contentsRect().width() - 2;
```

Comment:
If `setOuterLayout` hasn't been called before `outer_layout_` can't be used. So existing code using this class would segfault since it doesn't set the outer layout. So this needs to check if `outer_layout_` is set and conditionally either use the new code path or fallback to the old one.